### ISSUE TITLE
[Feature Request] Implement `__iter__` on PythonObject

### ISSUE THREAD

**Issue by xcloudplatform** [modularml/mojo#320]
### Review Mojo's priorities

- [X] I have read the [roadmap and priorities](https://docs.modular.com/mojo/roadmap.html#overall-priorities) and I believe this request falls within the priorities.

### What is your request?

```shell
when using python list it should not be of opaque type PythonObject. instead, it should be transparently bridged into list like object
```

### What is your motivation for this change?

```shell
%%python
package_list = [
    "nltk",
    "gensim",
    "fasttext",
    "spacy",
    "textblob",
    "pytorch_lightning",
    "fastai",
    "keras_tuner",
    "hyperopt",
    "pycaret",
    "shap",
    "eli5",
    "catboost",
    "xgboost",
    "lightgbm",
    "prophet",
    "opencv-python",
    "dlib",
    "imbalanced-learn",
    "tpot"
]




from PythonInterface import Python


available_packages = []
for package_name in package_list:
    try:
        Python.import_module(package_name)
        available_packages.append(package_name)
    except:
        pass
available_packages



error: Expression [14]:18:25: 'PythonObject' does not implement the '__iter__' method
    for package_name in package_list:
```

### Any other details?

```shell
x
```

**Comment by MainRo** [comment]
Is there a possible alternative for now?
For some tests I would like to iterate over a numpy array (in a for loop). 

```python
values = np.random.normal(0.0, 1.0, num_points)
for i in np.nditer(values):
    ...
```

**Comment by lattner** [comment]
@stumpOS can you take a look at this?

**Comment by lattner** [comment]
nice, please add to changelog if you haven't already!

**Comment by stumpOS** [comment]
Yep, PR coming soon
